core/src/test/java/org/elasticsearch/common/geo/GeoWKTShapeParserTests.java

```diff
@@ -242,40 +242,7 @@ public void testInvalidGeometryType() throws IOException {
 
     @Override
     public void testParseGeometryCollection() throws IOException {
-        int numShapes = randomIntBetween(0, 3);
-        GeometryCollectionBuilder builder = new GeometryCollectionBuilder();
-        Shape[] expected = new Shape[numShapes];
-        if (numShapes > 0) {
-            if (numShapes < 2) {
-                PolygonBuilder polygonBuilder = PolygonBuilder.class.cast(RandomShapeGenerator.createShape(random(),
-                    RandomShapeGenerator.ShapeType.POLYGON));
-                builder.shape(polygonBuilder);
-                expected[0] = jtsGeom(GEOMETRY_FACTORY.createPolygon(polygonBuilder.coordinates()[0][0]));
-            }
-
-            if (numShapes < 3) {
-                List<Coordinate> lineStringCoords = randomLineStringCoords();
-                LineStringBuilder lineStringBuilder = new LineStringBuilder(lineStringCoords);
-                builder.shape(lineStringBuilder);
-                expected[1] = jtsGeom(GEOMETRY_FACTORY.createLineString(lineStringCoords.toArray(
-                    new Coordinate[lineStringCoords.size()])));
-            }
-
-            if (numShapes < 4) {
-                List<Coordinate> multiPointCoords = randomLineStringCoords();
-                MultiPointBuilder multiPointBuilder = new MultiPointBuilder(multiPointCoords);
-                builder.shape(multiPointBuilder);
-                Shape[] points = new Shape[multiPointCoords.size()];
-                for (int i=0; i<multiPointCoords.size(); ++i) {
-                    Coordinate c = multiPointCoords.get(i);
-                    points[i] = SPATIAL_CONTEXT.makePoint(c.x, c.y);
-                }
-                expected[2] = shapeCollection(points);
-            }
-            assertExpected(shapeCollection(expected), builder);
-        } else {
-            // assert empty shape collection
-            assertEquals(shapeCollection(expected).isEmpty(), builder.build().isEmpty());
-        }
+        GeometryCollectionBuilder gcb = RandomShapeGenerator.createGeometryCollection(random());
```

Comment:
Great to have this, but it doesn't seem to test the no-shapes case. Would be nice to add this as another case (maybe wrapped in  `if (rarely())`, we don't need to test that edge case very often.